### PR TITLE
[SPARK-10060] [ML] [DOC] spark.ml DecisionTree user guide

### PR DESCRIPTION
**Pull request by jkbradley**
New user guide section ml-decision-tree.md, including code examples.

I have run all examples, including the Java ones.

CC: @manishamde @yanboliang @mengxr

### CODE REVIEW

**Comment by jkbradley** [comment]
Added WIP tag since I realized I still need to document some new features such as class probabilities.

File: docs/ml-decision-tree.md [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+
+// Load and parse the data file, converting it to a DataFrame.
+val data = MLUtils.loadLibSVMFile(sc, "data/mllib/sample_libsvm_data.txt").toDF()
+
+// Index labels, adding metadata to the label column.
+// Fit on whole dataset to include all labels in index.
+val labelIndexer = new StringIndexer()
+  .setInputCol("label")
+  .setOutputCol("indexedLabel")
+  .fit(data)
```
Comment (mengxr) [inline docs/ml-decision-tree.md]: minor: We could use `CrossValidator` to simplify the example. The pipeline becomes `[labelIndexer, vectorIndex, cv(dt, evaluator)]`. Just want to separate pipeline construction from fitting, no strong preference.

File: docs/ml-decision-tree.md [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+model = pipeline.fit(trainingData)
+
+# Make predictions.
+predictions = model.transform(testData)
+
+# Select example rows to display.
+predictions.select("prediction", "indexedLabel", "features").show(5)
+
+# Select (prediction, true label) and compute test error
+evaluator = MulticlassClassificationEvaluator(labelCol="indexedLabel", predictionCol="prediction", metricName="precision")
```
Comment (mengxr) [inline docs/ml-decision-tree.md]: line too wide?

File: docs/ml-decision-tree.md [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+  .fit(data);
+// Automatically identify categorical features, and index them.
+VectorIndexerModel featureIndexer = new VectorIndexer()
+  .setInputCol("features")
+  .setOutputCol("indexedFeatures")
+  .setMaxCategories(4) // features with > 4 distinct values are treated as continuous
+  .fit(data);
+
+// Split the data into training and test sets (30% held out for testing)
+DataFrame[] splits = data.randomSplit(new double[]{0.7, 0.3});
```
Comment (mengxr) [inline docs/ml-decision-tree.md]: space after `[]`

File: docs/ml-decision-tree.md (hunk 1 above)
Comment (jkbradley) [inline docs/ml-decision-tree.md]: I agree it's not ideal.  I'd prefer to leave CV out and to fix it in the next release once metadata are available for the prediction column.

**Comment by jkbradley** [comment]
Btw, what do you think of the table of input/output columns?  Too much?

**Comment by mengxr** [comment]
I think they are nice and clear:)

**Comment by jkbradley** [comment]
OK updated!

**Comment by jkbradley** [comment]
Note: I'll need to update the examples after [https://github.com/apache/spark/pull/8290] gets merged.

**Comment by mengxr** [comment]
LGTM. Merged into master and branch-1.5. We can update the examples in a separate PR. Thanks!

**Pull request closed without merging**